tests/components/metoffice/test_metoffice.py

```diff
@@ -0,0 +1,327 @@
+    "wind_gust": ("wind_gust", "16"),
+    "wind_speed": ("wind_speed", "9"),
+    "humidity": ("humidity", "50"),
+}
+
+
+class MockDateTime(datetime):
+    """Replacement for datetime that can be mocked for testing."""
+
+    def __new__(cls, *args, **kwargs):
+        """Override to just return base class."""
+        return datetime.__new__(datetime, *args, **kwargs)
+
+
+@patch("datetime.datetime", MockDateTime)
+async def test_one_weather_site_running(hass, requests_mock):
+    """Test the Met Office weather platform."""
+
+    # all metoffice test data encapsulated in here
+    mock_json = json.loads(load_fixture("metoffice.json"))
+    all_sites = json.dumps(mock_json["all_sites"])
+    wavertree_hourly = json.dumps(mock_json["wavertree_hourly"])
+
+    from datetime import datetime, timezone
+
+    MockDateTime.now = classmethod(
+        lambda *args, **kwargs: datetime(2020, 4, 25, 12, tzinfo=timezone.utc)
+    )
+
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text=all_sites)
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=3hourly", text=wavertree_hourly,
+    )
+
+    entry = MockConfigEntry(domain=DOMAIN, data=METOFFICE_CONFIG_WAVERTREE,)
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Wavertree weather platform expected results
+    entity = hass.states.get("weather.met_office_wavertree")
+    assert entity
+
+    assert entity.state == "sunny"
+    assert entity.attributes.get("temperature") == 17
+    assert entity.attributes.get("wind_speed") == 9
+    assert entity.attributes.get("wind_bearing") == "SSE"
+    assert entity.attributes.get("visibility") == "Good - 10-20"
+    assert entity.attributes.get("humidity") == 50
+
+
+@patch("datetime.datetime", MockDateTime)
+async def test_one_sensor_site_running(hass, requests_mock):
+    """Test the Met Office sensor platform."""
+
+    # all metoffice test data encapsulated in here
+    mock_json = json.loads(load_fixture("metoffice.json"))
+    all_sites = json.dumps(mock_json["all_sites"])
+    wavertree_hourly = json.dumps(mock_json["wavertree_hourly"])
+
+    from datetime import datetime, timezone
+
+    MockDateTime.now = classmethod(
+        lambda *args, **kwargs: datetime(2020, 4, 25, 12, tzinfo=timezone.utc)
+    )
+
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text=all_sites)
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=3hourly", text=wavertree_hourly,
+    )
+
+    entry = MockConfigEntry(domain=DOMAIN, data=METOFFICE_CONFIG_WAVERTREE,)
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    running_sensor_ids = hass.states.async_entity_ids("sensor")
+    assert len(running_sensor_ids) > 0
+    for running_id in running_sensor_ids:
+        sensor = hass.states.get(running_id)
+        _LOGGER.info(
```

Comment:
Please remove the logging.

Reply:
Done - was getting myself confused about what to expect, when, hence my logging it. Removed the logger.